src/controllers/accomodation.controller.js

```diff
@@ -70,6 +70,55 @@ class Accommodation {
     const query = await accomodationServices.bookAccommodation(req.user.id, req.body.accommodationId, req.body.roomId, req.body.departureDate, req.body.checkoutDate);
     return response.successMessage(res, 'Booking was successfully processed', 201, query);
   }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async rateAccomodation(req, res) {
+    try {
+      const { rate } = req.body;
+      const userId = req.user.id;
+      const { accomodationId } = req.params;
+
+      const rateData = { rate, userId, accomodationId };
+
+      const rateAccomodation = await accomodationServices.CreateAccomodationRate(rateData);
+      return response.successMessage(res, 'You rated this accomodation successfully', 201, rateAccomodation);
+    } catch (error) {
+      return response.errorMessage(res, error.message, 500);
+    }
+  }
+
+  /**
+   * This function helps user to rate accommodation which he booked.
+   * @param {Object} req request
+   * @param {*} res response
+   * @returns {Object} return accommodation response
+   */
+  static async GetAverageAccommodationRate(req, res) {
+    try {
+      const { accomodationId } = req.params;
+
+
+      let sum = 0;
+      const accomodations = await accomodationServices.checkAccommodationExistance(accomodationId);
```

Comment:
@mujohn26 How many accommodations? or you meant accommodation? Always check the spelling.
How if you check the existence of the accommodation in your middleware. That's the pattern I've seen the team adopting.

Reply:
I finished working on it.